yaksh/views.py

```diff
@@ -3191,3 +3191,70 @@ def download_course_progress(request, course_id):
     for student in stud_details:
         writer.writerow(student)
     return response
+
+
+@login_required
+@email_verified
+def course_forum(request, course_id):
+    user = request.user
+    course = get_object_or_404(Course, id=course_id)
+    posts = course.post.filter(active=True).order_by('-modified_at')
+    if request.method == "POST":
+        form = PostForm(request.POST, request.FILES)
+        if form.is_valid():
+            new_post = form.save(commit=False)
+            new_post.creator = user
+            new_post.course = course
+            new_post.save()
+            return redirect('yaksh:post_comments',
+                            course_id=course.id, uuid=new_post.uid)
+    else:
+        form = PostForm()
+    return render(request, 'yaksh/course_forum.html', {
+        'user': user,
+        'course': course,
+        'posts': posts,
+        'form': form,
+        'user': user
+        })
+
+
+@login_required
+@email_verified
+def post_comments(request, course_id, uuid):
+    user = request.user
+    post = get_object_or_404(Post, uid=uuid)
```

Comment:
Same here. Add a condition to check if the user is either course creator, teacher or student.